ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs

```diff
@@ -345,6 +344,7 @@ forkBlockProduction maxBlockSizeOverride IS{..} BlockProduction{..} =
       case mIsLeader of
         Nothing    -> return ()
         Just proof -> do
+          trace $ TraceForgeAboutToLead currentSlot
```

Comment:
Good catch.

Reply:
We rely on this event in the protocol tests. This is why CI is failing for this PR. 

I do admit that its name is a bit confusing. Maybe we can rename the existing trace point to `TraceForgeStartOfSlot` and use `TraceForgeAboutToLead` for the new location. /cc: @nfrisby

Reply:
The event needs to happen before the thread reads the ChainDB. Specifically, in the `ThreadNet` tests, the tracer itself (!!!) actually forges and adds an EBB before returning, which must be able to affect `(extLedger, (prevPoint, prevNo))`.

(... I absolutely should have added a comment here saying as much. :()

Reply:
Right. Do you agree that we should change its name to be slightly less misleading? The tests should of course also be updated.

Reply:
Sure. My thoughts on the name:

  * If we consider only its role in the tests, then something extremely concrete like `TraceForgeQueringChainDBTipBeforeForging` would be very telling.
  * If we prefer a more general constructor name, then `TraceForgeSlotOnset` (with comments at its definition and use sites about needing to precede the ChainDB query) might be more reasonable.

Reply:
Oops, I guess I got confused by the name too.

Reply:
FYI, @edsko renamed this tracer in PR #1459.